migration/process-migration/src/main/java/io/camunda/migration/process/adapter/os/OpensearchAdapter.java

```diff
@@ -0,0 +1,246 @@
+                                                    .value(FieldValue.of(PROCESSOR_STEP_ID))))))
+            .build();
+
+    final SearchResponse<ProcessorStep> searchResponse;
+
+    try {
+      searchResponse =
+          doWithRetry(
+              "Fetching last migrated process",
+              () -> client.search(request, ProcessorStep.class),
+              res -> res.timedOut() || Boolean.TRUE.equals(res.terminatedEarly()));
+    } catch (final Exception e) {
+      throw new MigrationException("Failed to fetch last migrated process", e);
+    }
+
+    return searchResponse.hits().hits().stream()
+        .map(Hit::source)
+        .filter(Objects::nonNull)
+        .map(ProcessorStep::getContent)
+        .findFirst()
+        .orElse(null);
+  }
+
+  @Override
+  public void writeLastMigratedEntity(final String processDefinitionKey) throws MigrationException {
+    final UpdateRequest<ProcessorStep, ProcessorStep> updateRequest =
+        new UpdateRequest.Builder<ProcessorStep, ProcessorStep>()
+            .index(migrationRepositoryIndex.getFullQualifiedName())
+            .id(PROCESSOR_STEP_ID)
+            .docAsUpsert(true)
+            .doc(upsertProcessorStep(processDefinitionKey))
+            .build();
+
+    try {
+      doWithRetry(
+          "Update last migrated process",
+          () -> client.update(updateRequest, ProcessorStep.class),
+          res -> res.result() == null);
+    } catch (final Exception e) {
+      throw new MigrationException("Failed to update migrated process", e);
+    }
+  }
+
+  @Override
+  public Set<ImportPositionEntity> readImportPosition() throws MigrationException {
+    final SearchRequest request =
+        new SearchRequest.Builder()
+            .index(importPositionIndex.getFullQualifiedName())
+            .query(
+                q ->
+                    q.wildcard(
+                        w -> w.field(ImportPositionIndex.ID).value("*-" + ProcessIndex.INDEX_NAME)))
+            .build();
+
+    final SearchResponse<ImportPositionEntity> searchResponse;
+
+    try {
+      searchResponse =
+          doWithRetry(
+              "Fetching import position",
+              () -> client.search(request, ImportPositionEntity.class),
+              res -> res.timedOut() || Boolean.TRUE.equals(res.terminatedEarly()));
+    } catch (final Exception e) {
+      throw new MigrationException("Failed to fetch import position", e);
+    }
+    return searchResponse.hits().hits().stream()
+        .map(Hit::source)
+        .filter(Objects::nonNull)
+        .collect(Collectors.toSet());
+  }
+
+  @Override
+  public void close() throws IOException {
+    client._transport().close();
+  }
+
+  @Override
+  public int getMaxRetries() {
+    return properties.getMaxRetries();
+  }
+
+  @Override
+  public Duration getMinDelay() {
+    return properties.getMinRetryDelay();
+  }
+
+  @Override
+  public Duration getMaxDelay() {
+    return properties.getMaxRetryDelay();
+  }
```

Comment:
❓ I think most of these methods we don't need here or? We could also simply use the properties in the Runner?